gst/droidcamsrc/gstdroidcamsrc.c

```diff
@@ -2635,3 +2662,41 @@ gst_droidcamsrc_post_preview (GstDroidCamSrc * src, GstSample * sample)
     gst_sample_unref (sample);
   }
 }
+
+static gboolean
+gst_droidcamsrc_update_jpeg_quality(GstDroidCamSrc * src)
+{
+  gboolean ret = FALSE;
+  const gchar* str = NULL;
+  gint jpeg_quality = 0;
+
+  GST_DEBUG_OBJECT (src, "update jpeg quality");
+
+  do {
+    g_rec_mutex_lock (&src->dev_lock);
+
+    if (!src->dev || !src->dev->params)
+      break;
+
+    jpeg_quality =
+      gst_droidcamsrc_params_get_int (src->dev->params, "jpeg-quality");
+
+    /*
+     * Value is not good, keep compatibility with current devices
+     * (-1 - value not present, 0 - value is not correct integer)
+     */
+    if (jpeg_quality <= 0)
+      break;
+
+    str = g_strdup_printf ("%u", src->jpeg_quality);
+    gst_droidcamsrc_params_set_string(src->dev->params, "jpeg-quality", str);
+    g_free(str);
+    ret = TRUE;
+  } while(0);
+
+  g_rec_mutex_unlock (&src->dev_lock);
+
+  GST_INFO_OBJECT (src, "setting jpeg quality to %u was %s", src->jpeg_quality,
```

Comment:
Minor language point, the 'was' doesn't really work. You'd need 'has' for FAILED. Best to not have it at all, which is fine for both cases e.g. "Setting jpeg quality to 10 OK" "Setting jpeg quality to 10 FAILED"